src/luascript.cpp

```diff
@@ -8740,43 +8740,67 @@ int LuaScriptInterface::luaPlayerSendOutfitWindow(lua_State* L)
 	return 1;
 }
 
-int LuaScriptInterface::luaPlayerAddMount(lua_State* L)
-{
-	// player:addMount(mountId)
+int LuaScriptInterface::luaPlayerAddMount(lua_State* L) {
+	// player:addMount(mountId or mountName)
 	Player* player = getUserdata<Player>(L, 1);
-	if (player) {
-		uint8_t mountId = getNumber<uint8_t>(L, 2);
-		pushBoolean(L, player->tameMount(mountId));
-	} else {
+	if (!player) {
 		lua_pushnil(L);
+		return 1;
 	}
+
+	uint8_t mountId;
+	if (isString(L, 2)) {
```

Comment:
Did you test this? This is making the method not work for number as isString uses lua_isstring and
```cpp
int lua_isstring (lua_State *L, int index);

Returns 1 if the value at the given acceptable index is a string or a number (which is always convertible to a string), and 0 otherwise
```